tools/create_service_bundle.py

```diff
@@ -0,0 +1,568 @@
+    return sdk_cmd.get_json_output(
+        "marathon app show {}".format(marathon_app_name), print_output=False
+    )
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def get_scheduler_tasks(app: dict) -> dict:
+    return list(
+        map(lambda task: get_tasks(task_id=task["id"])["tasks"][0], app["tasks"])
+    )
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def get_framework_id(package_name: str, service_name: str) -> str:
+    framework_ids = sdk_cmd.svc_cli(
+        package_name,
+        service_name,
+        "debug state framework_id",
+        json=True,
+        print_output=False,
+    )
+
+    assert len(framework_ids) == 1, "More than 1 Framework ID returned: {}".format(
+        framework_ids
+    )
+
+    return framework_ids[0]
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def debug_agent_files(agent_id: str) -> List[str]:
+    return sdk_cmd.cluster_request(
+        "GET", "/slave/{}/files/debug".format(agent_id)
+    ).json()
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def browse_agent_path(agent_id: str, agent_path: str) -> List[dict]:
+    return sdk_cmd.cluster_request(
+        "GET", "/slave/{}/files/browse?path={}".format(agent_id, agent_path)
+    ).json()
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def browse_executor_sandbox(agent_id: str, executor_sandbox_path: str) -> List[dict]:
+    return browse_agent_path(agent_id, executor_sandbox_path)
+
+
+def browse_executor_tasks(agent_id: str, executor_sandbox_path: str) -> List[dict]:
+    executor_sandbox = browse_executor_sandbox(agent_id, executor_sandbox_path)
+    tasks_directory = next(
+        filter(
+            lambda f: f["mode"].startswith("d") and f["path"].endswith("/tasks"),
+            executor_sandbox,
+        ),
+        None,
+    )
+
+    if tasks_directory:
+        return browse_agent_path(agent_id, tasks_directory["path"])
+    else:
+        return []
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def browse_task_sandbox(
+    agent_id: str, executor_sandbox_path: str, task_id: str
+) -> List[dict]:
+    executor_tasks = browse_executor_tasks(agent_id, executor_sandbox_path)
+
+    if executor_tasks:
+        task_sandbox_path = os.path.join(
+            executor_sandbox_path, "tasks/{}/".format(task_id)
+        )
+        return browse_agent_path(agent_id, task_sandbox_path)
+    else:
+        return []
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def download_agent_path(
+    agent_id: str, agent_file_path: str, output_file_path: str, chunk_size: int = 8192
```

Comment:
Does it make sense to set chunk size as a parameter?

Reply:
It already is a parameter?